app/cdap/services/DataFormatter/index.ts

```diff
@@ -59,6 +59,9 @@ export function dateTimeFormat(
   }
 ) {
   if (!(date instanceof Date)) {
+    if (typeof date === 'string') {
+      date = date.replace(/-/g, '/');
```

Comment:
The unit test for this helper was failing due to weird behaviour of JS Date constructor when dealing with strings containing "-" (moves back a day)